### PR TITLE
[opichuryn] sequence_task_2

### PR DESCRIPTION
**Pull request by opichuryn**

### CODE REVIEW

File: sequence_task_2/student/include/impl.h [hunk 1]
```diff
@@ -14,4 +14,9 @@
  * @note See https://en.cppreference.com/w/cpp/algorithm/rotate
  */
 template<class T, class BI>
-void rotate(std::list<T>& list, BI begin, BI new_begin, BI end);
+void rotate(std::list<T>& list, BI begin, BI new_begin, BI end)
+{
+    std::list<T> tmp;
```
Comment (smay1613) [inline sequence_task_2/student/include/impl.h:19]: why do you need "tmp" at all?

File: sequence_task_2/student/include/impl.h [hunk 2]
```diff
@@ -14,4 +14,9 @@
...
+    tmp.splice(tmp.end(), list, begin, new_begin);
```
Comment (smay1613) [inline sequence_task_2/student/include/impl.h:20]: please use only iterators from input

File: sequence_task_2/student/include/impl.h (hunk 1 above)
Comment (opichuryn) [reply sequence_task_2/student/include/impl.h]: Done!

File: sequence_task_2/student/include/impl.h (hunk 2 above)
Comment (opichuryn) [reply sequence_task_2/student/include/impl.h]: Done!

**Pull request closed without merging**